File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    lenn = len(numbers)
    ans = 0
    
    def dfs(i, summ) :
        nonlocal ans
        if i == lenn :
            if summ == target :
                ans += 1
            return
        
        dfs(i+1, summ + numbers[i])
        dfs(i+1, summ - numbers[i])
    
    dfs(0,0)
    
    return ans

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    public int solution(int[] numbers, int target) {
        return dfs(numbers, target, 0, 0);
    }
    
    public int dfs(int[] arr, int t, int idx, int sum) {
        // 기저 조건
        if (idx >= arr.length) {
            return (sum == t) ? 1 : 0;
        }
        
        // 재귀
        return dfs(arr, t, idx+1, sum+arr[idx]) + dfs(arr,t,idx+1,sum-arr[idx]);
    }
}